base/sparse/cholmod.jl

```diff
@@ -1603,6 +1603,9 @@ Ac_ldiv_B(L::FactorComponent, B::RowVector) = ctranspose(L)\B # ambiguity
 # linalg/factorizations.jl
 (\){T<:VTypes}(L::Factor{T}, b::StridedVector) = Vector(L\convert(Dense{T}, b))
 (\){T<:VTypes}(L::Factor{T}, B::StridedMatrix) = Matrix(L\convert(Dense{T}, B))
+# nix ambiguities with (\){T<:BlasReal}(F::Factorization{T}, V::VecOrMat{Complex{T}})
+(\){T<:Float64}(L::Factor{T}, B::Vector{Complex{T}}) = Vector(L\convert(Dense{T}, B))
```

Comment:
I shudder to think of what this would mean in practice, but for functions that accept array inputs, I'm beginning to wonder if we should become more systematic in declaring that "container types rule" and that we should get away from dispatching on specific eltypes at the same level of the dispatch hierarchy. Sketch:
- `\` dispatches on container type
- `\` has a fallback definition `\(L, B) = _ldiv_eltype(L, B)`
- The *only* thing you're allowed to dispatch on in `_ldiv_eltype` is the element type, i.e., all methods look like `_ldiv_eltype{T<:BlasFloat}(L::AbstractMatrix{T}, B::AbstractVecOrMat{T}) = call_BLAS_method(convert(Matrix{T}, L), convert(VecOrMat{T}, B))`.